gridscale/relation-manager/serverRelationManager.go

```diff
@@ -0,0 +1,377 @@
+package relation_manager
+
+import (
+	"context"
+	"fmt"
+	"github.com/gridscale/gsclient-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"net/http"
+)
+
+//firewallRuleTypes defines all types of firewall rules
+var firewallRuleTypes = []string{"rules_v4_in", "rules_v4_out", "rules_v6_in", "rules_v6_out"}
+
+//ServerRelationManger is an wrapper of gsclient which is used for
+//managing relations of a server in gridscale terraform provider
+type ServerRelationManger struct {
+	gsc  *gsclient.Client
+	data *schema.ResourceData
+}
+
+//NewServerRelationManger creates a new instance ServerRelationManger
+func NewServerRelationManger(gsc *gsclient.Client, d *schema.ResourceData) *ServerRelationManger {
+	return &ServerRelationManger{gsc, d}
+}
+
+//getGSClient returns gsclient from server relation manager
+func (c ServerRelationManger) getGSClient() *gsclient.Client {
+	return c.gsc
+}
+
+//getData returns resource data from server relation manager
+func (c ServerRelationManger) getData() *schema.ResourceData {
+	return c.data
+}
+
+//LinkStorages links storages to a server
+//**Note: if no storage is set as the boot device. The storage in the
+//1st order of the list will be set as the boot device automatically
```

Comment:
Consider rewording:
- "The storage in the 1st order of the list will be set as the boot device automatically" to 
+ "The first storage in the list will be automatically set as the boot device"